lib/configproxy.js

```diff
@@ -528,72 +532,77 @@ class ConfigurableProxy extends EventEmitter {
+        args.push({ target: target });
 
-      // dispatch the actual method, either:
-      // - proxy.web(req, res, options, errorHandler)
-      // - proxy.ws(req, socket, head, options, errorHandler)
-      that.proxy[kind].apply(that.proxy, args);
+        // add error handling
+        args.push(function (e) {
+          that.handleProxyError(503, kind, req, res, e);
+        });
 
-      // update timestamp on any request/reply data as well (this includes websocket data)
-      req.on("data", function () {
-        that.updateLastActivity(prefix);
-      });
+        // dispatch the actual method, either:
+        // - proxy.web(req, res, options, errorHandler)
+        // - proxy.ws(req, socket, head, options, errorHandler)
+        that.proxy[kind].apply(that.proxy, args);
 
-      res.on("data", function () {
-        that.updateLastActivity(prefix);
-      });
+        // update timestamp on any request/reply data as well (this includes websocket data)
+        req.on("data", function () {
+          that.updateLastActivity(prefix);
+        });
 
-      if (kind === "web") {
-        // update last activity on completion of the request
-        // only consider 'successful' requests activity
-        // A flood of invalid requests such as 404s or 403s
-        // or 503s because the endpoint is down
-        // shouldn't make it look like the endpoint is 'active'
-
-        // we no longer register activity at the *start* of the request
-        // because at that point we don't know if the endpoint is even available
-        res.on("finish", function () {
-          // (don't count redirects...but should we?)
-          if (res.statusCode < 300) {
-            that.updateLastActivity(prefix);
-          } else {
-            that.log.debug("Not recording activity for status %s on %s", res.statusCode, prefix);
-          }
+        res.on("data", function () {
+          that.updateLastActivity(prefix);
         });
-      }
-    });
+
+        if (kind === "web") {
+          // update last activity on completion of the request
+          // only consider 'successful' requests activity
+          // A flood of invalid requests such as 404s or 403s
+          // or 503s because the endpoint is down
+          // shouldn't make it look like the endpoint is 'active'
+
+          // we no longer register activity at the *start* of the request
+          // because at that point we don't know if the endpoint is even available
+          res.on("finish", function () {
+            // (don't count redirects...but should we?)
+            if (res.statusCode < 300) {
+              that.updateLastActivity(prefix);
+            } else {
+              that.log.debug("Not recording activity for status %s on %s", res.statusCode, prefix);
+            }
+          });
+        }
+      })
+      .catch(function (e) {
+        if (res.finished) throw e;
+        that.handleProxyError(500, kind, req, res, e);
+      });
```

Comment:
These lines are the only changes in this function: adding the catch. prettier chooses a different indentation when there's both a `.then` and a `.catch`, which makes this PR look bigger than it is.